Lambdas/GHOAuth/OAuthLambda.swift

```diff
@@ -0,0 +1,196 @@
+    typealias Output = APIGatewayV2Response
+
+    let client: HTTPClient
+    let logger: Logger
+    let secretsRetriever: SecretsRetriever
+    let jsonDecoder: JSONDecoder
+    let jsonEncoder: JSONEncoder
+    let userService: UserService
+
+    /// We don't do this in the initializer to avoid a possible unnecessary
+    /// `secretsRetriever.getSecret()` call which costs $$$.
+    var discordClient: any DiscordClient {
+        get async throws {
+            let botToken = try await secretsRetriever.getSecret(arnEnvVarKey: "BOT_TOKEN_ARN")
+            return await DefaultDiscordClient(httpClient: client, token: botToken)
+        }
+    }
+
+    init(context: LambdaInitializationContext) async throws {
+        self.client = HTTPClient(eventLoopGroupProvider: .shared(context.eventLoop))
+        self.logger = context.logger
+
+        let awsClient = AWSClient(httpClientProvider: .shared(client))
+        self.secretsRetriever = SecretsRetriever(awsClient: awsClient, logger: logger)
+
+        self.userService = UserService(awsClient, logger)
+
+        self.jsonDecoder = JSONDecoder()
+        self.jsonEncoder = JSONEncoder()
+    }
+
+    func handle(_ event: APIGatewayV2Request, context: LambdaContext) async -> APIGatewayV2Response {
+        logger.debug("Received event: \(event)")
+
+        guard let code = event.queryStringParameters?["code"] else {
+            return .init(statusCode: .badRequest, body: "Missing code query parameter")
+        }
+
+        let accessToken: String
+
+        do {
+            accessToken = try await getGHAccessToken(code: code)
+        } catch {
+            logger.error("Error getting access token", metadata: [
+                "error": "\(error)"
+            ])
+            return .init(statusCode: .badRequest, body: "Error getting access token")
+        }
+
+        let user: User
+
+        do {
+            user = try await getGHUser(accessToken: accessToken)
+        } catch {
+            logger.error("Error getting user ID", metadata: [
+                "error": "\(error)"
+            ])
+            return .init(statusCode: .badRequest, body: "Error getting user")
+        }
+
+        let jwt: GHOAuthPayload
+
+        do {
+            jwt = try await verifyState(state: String(event.queryStringParameters?["state"] ?? ""))
+        } catch {
+            logger.error("Error during state verification", metadata: [
+                "error": "\(error)",
+                "state": .string(event.queryStringParameters?["state"] ?? "")
+            ])
+            return .init(statusCode: .badRequest, body: "Error verifying state")
+        }
+
+        do {
+            try await userService.linkGithubID(discordID: jwt.discordID.rawValue, githubID: "\(user.id)")
+        } catch {
+            logger.error("Error linking user to GitHub account", metadata: [
+                "discordID": .string(jwt.discordID.rawValue),
+                "githubID": .string("\(user.id)")
```

Comment:
both values conform to `CustomStringConvertible` so this can be simplified to:
```suggestion
                "discordID": .stringConvertible(jwt.discordID),
                "githubID": .stringConvertible(user.id)
```